worlds/zillion/patch.py

```diff
@@ -44,15 +50,17 @@ def read_contents(self, opened_zipfile: zipfile.ZipFile) -> None:
         super().read_contents(opened_zipfile)
         self.gen_data_str = opened_zipfile.read("gen_data.json").decode()
 
+    @override
     def patch(self, target: str) -> None:
         self.read()
         write_rom_from_gen_data(self.gen_data_str, target)
 
 
 def get_base_rom_path(file_name: Optional[str] = None) -> str:
-    options = Utils.get_options()
+    from . import ZillionSettings, ZillionWorld
+    settings: ZillionSettings = ZillionWorld.settings
```

Comment:
I'm assuming you have checked it, but shouldn't the type deduction work here?

Reply:
Actually, it shouldn't work, even if it does.
Did you know you're relying on a bug in PyCharm here?

What will this code print?
And what does PyCharm say it will print (when you put your mouse over `something` in the `print`)?
```python
class M(type):
    @property
    def something(self) -> int:
        return 5


class C(metaclass=M):
    something: str = "hello"


print(C.something)
```

Reply:
If the meta class screws with the members, that does not change the static type annotation of the members.

Reply:
The metaclass gives a **static** type annotation that is correct.
Pyright sees the correct type of `int`.
PyCharm is giving incorrect information.

Reply:
Oh, as usual I am wrong. mypy is smart enough to understand that property. We should maybe fix this by moving the property to `__new__` then.

Reply:
<details><summary>...</summary>I'm not convinced that metaclasses are the best way to do anything that we're doing with them.</details>

Reply:
I think our model is ok, even if it's a bit too much magic.
Generally speaking, the concept of having a class-like definition that ends up being a different type during runtime is fine.

I feel like hiding the implementation from static analyzers is the only way for that to work if they become "too smart".

Also interestingly
```sh
$ cat test.py
from worlds.soe import SoEWorld

reveal_type(SoEWorld.settings)

$ python -m mypy test.py 
test.py:3: note: Revealed type is "worlds.soe.SoESettings"

$ python -m pyright test.py
[...] information: Type of "SoEWorld.settings" is "Any"
```

Reply:
Yeah, it's not bad as it is.